Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -497,7 +500,60 @@ fileprivate extension Array<AnnotatedTestItem> {
   }
 }
 
+fileprivate extension Array<TestItem> {
+  /// If multiple testItems share the same ID we add more context to make it unique.
+  /// Two tests can share the same ID when two swift testing tests accept
+  /// arguments of different types, i.e:
+  /// ```
+  /// @Test(arguments: [1,2,3]) func foo(_ x: Int) {}
+  /// @Test(arguments: ["a", "b", "c"]) func foo(_ x: String) {}
+  /// ```
+  ///
+  /// or when tests are in separate files but don't conflict because they are marked
+  /// private, i.e:
+  /// ```
+  /// File1.swift: @Test private func foo() {}
+  /// File2.swift: @Test private func foo() {}
+  /// ```
+  ///
+  /// If we encounter one of these cases, we need to deduplicate the ID
+  /// by appending `/filename:filename:lineNumber`.
+  func deduplicateIds() -> [TestItem] {
+    var idCounts: [String: Int] = [:]
+    var result: [TestItem] = []
```

Comment:
Not the most important thing but you can avoid having to grow `result` by adding 

```swift
result.reserveCapacity(self.count)
```

Assuming that most tests won’t need to be de-duped.

Reply:
I'm wondering, is this preferable to returning the results of a `self.map { ...`